### ISSUE TITLE
iwlwifi firmware fails to load after several suspends to RAM

### ISSUE THREAD

**Issue by schmittlauch** [NixOS/nixpkgs#58482]
## Issue description

Since several week the iwlwifi firmware for my Intel Corporation Wireless 7260 (rev 83) card fails to load after resuming from a suspend RAM.
As at boot the firmware is always loaded successfully, a reboot temporarily fixes the issue.
 
The issue doesn't appear after each suspen-resume cycle, but after several tries you'll run into it with a very high probability. It seems to have happened once even without suspending and resuming, but only with disabling and enabling the WLAN adapter (soft rfkill).

The issue appeared both with kernel 4.14.107 and 4.19.31

Here are the logs:

journald log:

```
Mär 28 12:24:28 thinknix wpa_supplicant[1492]: rfkill: WLAN unblocked
Mär 28 12:24:29 thinknix kernel: iwlwifi 0000:03:00.0: Could not load the [0] uCode section
Mär 28 12:24:29 thinknix kernel: iwlwifi 0000:03:00.0: Failed to start INIT ucode: -5
Mär 28 12:24:29 thinknix kernel: iwlwifi 0000:03:00.0: Failed to run INIT ucode: -5
Mär 28 12:24:29 thinknix kernel: iwlwifi 0000:03:00.0: Failed to start RT ucode: -5
Mär 28 12:24:30 thinknix NetworkManager[1212]: <warn>  [1553772270.9995] platform-linux: do-change-link[3]: failure changing link: failure 5 (Input/output error)
```

dmesg |grep wifi 

```
[   14.476997] iwlwifi 0000:03:00.0: loaded firmware version 17.3216344376.0 op_mode iwlmvm
[   14.578151] iwlwifi 0000:03:00.0: Detected Intel(R) Dual Band Wireless AC 7260, REV=0x144
[   14.600870] iwlwifi 0000:03:00.0: base HW address: <censored>
[   14.873952] iwlwifi 0000:03:00.0 wlp3s0: renamed from wlan0
```

### Steps to reproduce

1. Boot NixOS on a system with an Intel 7260 WLAN card (in my case a ThinkPad T440S)
1. Connect to a WLAN network (in my case using NetworkManager)
1. suspend to RAM and resume, alternatively: disable and enable WLAN using rfkill
1. observe whether it is possible to reconnect to a WLAN network
1. if issue does not appear, goto 3

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.107, NixOS, 18.09.2372.f4b2ae96cc8 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(root): `"nixos-18.09.2394.2d656a97297, nixos-unstable-19.09pre173166.373488e6f4c"`
 - channels(user): `"home-manager-18.09"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by jcumming** [comment]
Similar things happen with my Intel 8260 AC card, although not often enough I've looked for upstream bug reports.

**Comment by schmittlauch** [comment]
I reported that bug upstream https://bugzilla.kernel.org/show_bug.cgi?id=203521

The advice was to try kernel 5.1, I'll do that once it lands in release-19.03.

/cc @makefoo

**Comment by makefu** [comment]
@schmittlauch thanks! I will try 5.1 as soon as it is available in 19.03 as well :)

**Comment by fpletz** [comment]
We are now on 5.2 on 19.03. If the problem persists please reopen. Thanks!
